proc-macros/src/render_client.rs

```diff
@@ -83,6 +113,7 @@ impl RpcDescription {
 		// `returns` represent the return type of the *rust method* (`Result< <..>, jsonrpsee::core::Error`).
 		let (called_method, returns) = if let Some(returns) = &method.returns {
 			let called_method = quote::format_ident!("request");
+			let returns = self.patch_result_error(returns);
```

Comment:
I don't understand what this `patch_result_error` function is doing now offhand; could you explain it to me?

Reply:
Sure! I'll also add the comment documenting it since it's not trivial!

It tries to rewrite the return type from `Result<_, E>` (generic `E`) to `Result<_, Error>` (concrete `Error`) - which is required because the client can't be expected to parse the custom error type back from JSON.